cpp/src/arrow/ipc/writer.h

```diff
@@ -418,5 +344,76 @@ Status GetSparseTensorPayload(const SparseTensor& sparse_tensor, MemoryPool* poo
 
 }  // namespace internal
 
+// Deprecated functions
+
+/// \class RecordBatchStreamWriter
+/// \brief Synchronous batch stream writer that writes the Arrow streaming
+/// format
+class ARROW_EXPORT RecordBatchStreamWriter : public RecordBatchWriter {
+ public:
+  /// Create a new writer from stream sink and schema. User is responsible for
+  /// closing the actual OutputStream.
+  ///
+  /// \param[in] sink output stream to write to
+  /// \param[in] schema the schema of the record batches to be written
+  /// \param[out] out the created stream writer
+  /// \return Status
+  ARROW_DEPRECATED("Use Result-returning version")
```

Comment:
We need to update this message because we don't have Result-returning version `Open`.

Reply:
OK

Reply:
I've restored the Result-returning version `Open` because it's provided since 0.16.0.
Keeping it keeps backward compatibility.